pydantic_ai_slim/pydantic_ai/models/function.py

```diff
@@ -127,7 +128,9 @@ class FunctionAgentModel(AgentModel):
     stream_function: StreamFunctionDef | None
     agent_info: AgentInfo
 
-    async def request(self, messages: list[Message]) -> tuple[ModelAnyResponse, result.Cost]:
+    async def request(
+        self, messages: list[Message], model_settings: ModelSettings | None = None
```

Comment:
I think you want to add `model_settings` to the `AgentInfo` argumment.

Reply:
I'm not following that — all of the other models have it on this method, which makes sense to me, and that's why the signature was updated. But the other models don't have an `AgentInfo` field.

(I'm assuming Sydney will be confused by this comment in the same way I am.)

Reply:
Done, thanks!